digdag-core/src/main/java/io/digdag/core/log/LocalFileLogServerFactory.java

```diff
@@ -193,6 +217,24 @@ public void close()
             {
                 try {
                     output.close();
+
+                    if (realtimeFollow) {
+                        // Delaying for deterrence duplicate output in digdag log --follow
+                        try {
+                            Thread.sleep(1000);
+                        }
+                        catch (InterruptedException ex) {
+                            Thread.currentThread().interrupt();
+                        }
+
+                        try (InputStream in = Files.newInputStream(temporaryPath);
+                             OutputStream out = new GZIPOutputStream(Files.newOutputStream(path, CREATE, APPEND), 16*1024)) {
+                            ByteStreams.copy(in, out);
+                        }
+                        finally {
+                            temporaryPath.toFile().delete();
```

Comment:
https://github.com/treasure-data/digdag/pull/700#discussion_r176675750

> Yes remained, should we retrying deletion?

No. How about just deleting it in `finally` block like this?

```java
     if (realtimeFollow) {
         // Delaying for deterrence duplicate output in digdag log --follow
         try {
             Thread.sleep(1000);
         }
         catch (InterruptedException ex) {
             Thread.currentThread().interrupt();
         }

         try {
             output.close();

             try (InputStream in = Files.newInputStream(temporaryPath);
                 OutputStream out = new GZIPOutputStream(Files.newOutputStream(path, CREATE, APPEND), 16*1024)) {
                 ByteStreams.copy(in, out);
             }
         }
         finally {
             temporaryPath.toFile().delete();
         }
    }
    else {
         output.close();
    }
```

Reply:
@komamitsu How catch IOException from `output.close();`?

Reply:
How is this?

```java
try {
    output.close();

    if (realtimeFollow) {
        // Delaying for deterrence duplicate output in digdag log --follow
        try {
            Thread.sleep(1000);
        }
        catch (InterruptedException ex) {
            Thread.currentThread().interrupt();
        }

        try (InputStream in = Files.newInputStream(temporaryPath);
             OutputStream out = new GZIPOutputStream(Files.newOutputStream(path, CREATE, APPEND), 16*1024)) {
            ByteStreams.copy(in, out);
        }
    }
}
catch (IOException ex) {
    throw Throwables.propagate(ex);
}
finally {
    if (realtimeFollow) {
        temporaryPath.toFile().delete();
    }
}
```

Reply:
@komamitsu Fixed it :D